apps/spruce/src/pages/image/PackagesTable/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { useMemo, useRef, useState } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Pagination from "@leafygreen-ui/pagination";
+import {
+  useLeafyGreenTable,
+  LGColumnDef,
+  ColumnFiltersState,
+} from "@leafygreen-ui/table";
+import { useParams } from "react-router-dom";
+import { SettingsCard, SettingsCardTitle } from "components/SettingsCard";
+import { BaseTable } from "components/Table/BaseTable";
+import { slugs } from "constants/routes";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import {
+  Package,
+  ImagePackagesQuery,
+  ImagePackagesQueryVariables,
+} from "gql/generated/types";
+import { IMAGE_PACKAGES } from "gql/queries";
+import { useFirstImage } from "hooks";
+
+export const PackagesTable: React.FC = () => {
+  const { [slugs.imageId]: imageId } = useParams();
+  const { image: firstImage } = useFirstImage();
+  const selectedImage = imageId ?? firstImage;
+  const dispatchToast = useToastContext();
+  const [pagination, setPagination] = useState({
+    pageIndex: 0,
+    pageSize: 10,
+  });
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const { data: packagesData } = useQuery<
+    ImagePackagesQuery,
+    ImagePackagesQueryVariables
+  >(IMAGE_PACKAGES, {
+    variables: {
+      imageId: selectedImage,
+      opts: {
+        page: pagination.pageIndex,
+        limit: pagination.pageSize,
+        name: columnFilters.find((filter) => filter.id === "name")
+          ?.value as string,
+      },
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error loading image packages: ${err.message}`,
+      );
+    },
+  });
+
+  const packageEntries = useMemo(
+    () => packagesData?.image?.packages.data ?? [],
+    [packagesData?.image?.packages.data],
+  );
+
+  const totalNumPackages = useMemo(
+    () => packagesData?.image?.packages.totalCount ?? 0,
+    [packagesData?.image?.packages.totalCount],
+  );
+
+  const columns: LGColumnDef<Package>[] = [
+    {
+      header: "Name",
+      accessorKey: "name",
+      enableColumnFilter: true,
+      filterFn: "includesString",
+      meta: {
+        search: {
+          "data-cy": "package-name-filter",
+          placeholder: "Search name",
+        },
+        width: "15%",
+      },
+    },
+    {
+      header: "Manager",
+      accessorKey: "manager",
+    },
+    {
+      header: "Version",
+      accessorKey: "version",
+    },
+  ];
+
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const table = useLeafyGreenTable<Package>({
+    columns,
+    data: packageEntries,
+    containerRef: tableContainerRef,
+    defaultColumn: {
+      enableColumnFilter: false,
+    },
+    manualPagination: true,
+    manualFiltering: true,
+    rowCount: totalNumPackages,
```

Comment:
just curious, does this `rowCount` do anything? It seems like it's handled by `Pagination`

Reply:
The docs seem to suggest passing it in if we know the rowCount because it makes calculating the total number of entries easier for tanstack.
[Docs](https://tanstack.com/table/latest/docs/guide/pagination#pagination-row-model)